src/components/pool-display.tsx

```diff
@@ -44,43 +47,53 @@ export function PoolDisplay({ poolId = 0n }: PoolDisplayProps) {
   const userInfo = useReadContract({
     ...stakingContract,
     functionName: "getStakingUserInfo",
-    args: [poolId, account.address ?? "0x0"],
+    args: [poolId, address ?? "0x0"],
     query: {
-      enabled: account.isConnected && !!account.address,
+      enabled: isConnected && !!address,
     },
   });
+  
   const pendingRewards = useReadContract({
     ...stakingContract,
     functionName: "getPendingStakingRewards",
-    args: [poolId, account.address ?? "0x0"],
+    args: [poolId, address ?? "0x0"],
     query: {
-      enabled: account.isConnected && !!account.address,
+      enabled: isConnected && !!address,
     },
   });
 
   const allowance = useReadContract({
     abi: erc20Abi,
     address: lpTokenAddress,
     functionName: "allowance",
-    args: [account.address ?? "0x0", stakingContract.address],
+    args: [address ?? "0x0", stakingContract.address],
     query: {
-      enabled: account.isConnected && !!account.address && !!lpTokenAddress,
+      enabled: isConnected && !!address && !!lpTokenAddress,
     },
   });
+  
   const balance = useReadContract({
     abi: erc20Abi,
     address: lpTokenAddress,
     functionName: "balanceOf",
-    args: [account.address ?? "0x0"],
+    args: [address ?? "0x0"],
     query: {
-      enabled: account.isConnected && !!account.address && !!lpTokenAddress,
+      enabled: isConnected && !!address && !!lpTokenAddress,
     },
   });
 
   const onSuccess = async (hash: `0x${string}`) => {
-    if (!publicClient) return;
+    if (!publicClient) {
+      setErrorMessage("No public client available");
+      return;
+    }
+    
     try {
-      const receipt = await waitForTransactionReceipt(publicClient, { hash });
+      const receipt = await waitForTransactionReceipt(publicClient, { 
+        hash,
+        timeout: 60_000,
```

Comment:
Perhaps it makes sense to decrease to 15 seconds if a block is around that time?
```suggestion
        timeout: 15_000,
```